numbat-cli/src/main.rs

```diff
@@ -201,35 +200,45 @@ impl Cli {
                 .load_currency_module_on_demand(true);
         }
 
-        let code_and_source: Option<(String, CodeSource)> = if let Some(ref path) = self.file {
-            Some((
-                fs::read_to_string(path).context(format!(
+        let mut code_and_source = Vec::new();
+
+        if let Some(ref path) = self.file {
+            code_and_source.push((
+                (fs::read_to_string(path).context(format!(
                     "Could not load source file '{}'",
                     path.to_string_lossy()
-                ))?,
+                ))?),
                 CodeSource::File(path.clone()),
-            ))
-        } else {
-            self.expression
-                .as_ref()
-                .map(|exprs| (exprs.iter().join("\n"), CodeSource::Text))
+            ));
         };
 
-        if let Some((code, code_source)) = code_and_source {
-            let result = self.parse_and_evaluate(
-                &code,
-                code_source,
-                ExecutionMode::Normal,
-                self.config.pretty_print,
-            );
+        if let Some(expressions) = &self.expression {
+            code_and_source.push((expressions.iter().join("\n"), CodeSource::Text));
+        }
 
-            match result {
-                std::ops::ControlFlow::Continue(()) => Ok(()),
-                std::ops::ControlFlow::Break(ExitStatus::Success) => Ok(()),
-                std::ops::ControlFlow::Break(ExitStatus::Error) => {
-                    bail!("Interpreter stopped due to error")
-                }
+        if !code_and_source.is_empty() {
+            let mut code_result = Ok(());
+
+            for (code, code_source) in code_and_source {
+                let result = self.parse_and_evaluate(
+                    &code,
+                    code_source,
+                    ExecutionMode::Normal,
+                    self.config.pretty_print,
+                );
+
+                let result_status = match result {
+                    std::ops::ControlFlow::Continue(()) => Ok(()),
+                    std::ops::ControlFlow::Break(ExitStatus::Success) => Ok(()),
```

Comment:
I was not quite sure if this case should result in an error/warning for `FILE` if both `FILE` and `CODE` are given?

Reply:
Good catch. Is `ControlFlow::Break(ExitStatus::Success)` even a possible? :-)

Reply:
I don't think so.  `Cli.parse_and_evaluate()` can only return `ControlFlow::Continue(())` or `ControlFlow::Break(ExitStatus::Error)`.

Actually I don't see  `ControlFlow::Break(ExitStatus::Success)`  being used anywhere, in which cases was initially intended to be used?

Reply:
I guess I intended it to handle things like a hypothetical `exit(0)` call from a Numbat program. Let's remove it for now!